docs/tutorials/writing-a-drain-plugin.md

```
@@ -0,0 +1,816 @@
+```go
+type DrainRequestStatus struct {
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+### The completion contract
+
+node-drainer waits for a `status.conditions[]` entry whose **Type** and **Status** match its
+configuration. The convention (and what this tutorial uses) is:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)